app/assets/javascripts/i18n.js

```diff
@@ -772,6 +772,10 @@
 
     options = this.prepareOptions(options, DATE);
 
+    if (isNaN(date.getTime())) {
+      throw new Error('I18n.strftime() requires a valid date object, but received an invalid date.');
```

Comment:
Yes, `Error` constructor is one of the global object that all the JS environments (ancient browsers, Node.js, Electron etc.) have.